enhancements/installer/aws-ingress-nlb-security-group.md

```diff
@@ -0,0 +1,501 @@
+	// when using load balancer type Network Load Balancer (NLB).
+	// When this field is enabled with LBType NLB, all ingresscontrollers (including the
+	// default ingresscontroller) will be created using security group in the NLBs
+	// by default.
+	//
+	// If this field is not set explicitly, it defaults to no security groups in the NLB.
+	// This default is subject to change over time.
+	//
+	// +optional
+	SecurityGroupEnabled bool `json:"securityGroupEnabled,omitempty"`
+}
+```
+
+- TBD: deprecate the field `platform.aws.lbType` in favor of `platform.aws.ingressController.loadBalancerType`
+
+#### ROSA Classic
+
+- TBD: API changes in Hive to read and process the new install-config option.
+
+#### Hypershift/ROSA HCP
+
+- TBD: API changes in Hypershift to configure CIO based on the install-config option.
+
+### Topology Considerations
+
+#### Hypershift / Hosted Control Planes
+
+- TODO/TBD: The flow needs to be explored to determine how the install-config option propagates to the Hosted Control Plane's CIO configuration. This might involve setting the annotation on the Service created in the guest cluster.
+
+#### Standalone Clusters
+
+<!-- Is the change relevant for standalone clusters? -->
+
+> TODO/TBD: All changes is proposed initially and exclusively for Standalone clusters.
+
+
+#### Single-node Deployments or MicroShift
+
+> TODO/TBD
+
+
+### Implementation Details/Notes/Constraints
+
+> WIP/TBReviewed
+
+- The implementation in CCM should handle the case where the `service.beta.kubernetes.io/aws-load-balancer-managed-security-group` annotation is set to `true` but the service type is not `NLB` (`aws-load-balancer-type: nlb`). In this scenario, the CCM should likely log a warning mentioning the annotation is supported only on NLB.
+- The initial implementation will focus on creating a single Security Group per NLB.
+- The Security Group naming convention should be consistent and informative, including the cluster ID, namespace, and service name to aid in identification and management in the AWS console. (TODO: need review, perhaps we just create the same name as LB?)
+- Egress rule management in CCM needs careful consideration to avoid overly permissive rules. The initial implementation should restrict egress to the necessary ports and protocols for communication with the backend pods (traffic ports and health check ports) within the cluster's VPC.
+- Proper IAM permissions for the CCM's service account will be required to allow it to create, describe, and delete Security Groups in AWS. This needs to be documented as a prerequisite. (TODO review: maybe we don't need this as IAM permissions would be already granted for CLB [review it])
+
+### Risks and Mitigations
+
+> WIP/TBReviewed
+
+- **Increased complexity in CCM**: Adding security group management to CCM increases its complexity. Mitigation: Focus on a minimal and well-tested implementation, drawing inspiration from the existing CLB security group management logic in CCM.   
```

Comment:
What if we just allowed it to attach existing security groups? And didn't have the default managed security group?

Reply:
BYO SG added as a later phase.